File: app/src/androidTest/java/com/agoda/sneakershop/SneakerDetailTest.java
package com.agoda.sneakershop;

import com.agoda.sneakershop.screen.sneaker.SneakerActivity;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

/**
 * Created by vwongsawangt on 10/5/2017 AD.
 */

@RunWith(AndroidJUnit4.class)
public class SneakerDetailTest {

    @Rule
    public ActivityTestRule<SneakerActivity> mActivityRule = new ActivityTestRule<>(SneakerActivity.class);

    @Test
    public void checkFirstItemDetail() {
        //TODO: Check the first item name and category
    }

    @Test
    public void checkLastItemDetail() {
        //TODO: Check the last item name and category
    }

}

File: app/src/androidTest/kotlin/com/agoda/sneakershop/SneakerDetailTest.kt
package com.agoda.sneakershop

import android.support.test.rule.ActivityTestRule
import android.support.test.runner.AndroidJUnit4
import com.agoda.sneakershop.screen.sneaker.SneakerActivity
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

/**
 * Created by vwongsawangt on 10/6/2017 AD.
 */
@RunWith(AndroidJUnit4::class)
class SneakerDetailTest {

    @JvmField
    @Rule
    val rule = ActivityTestRule(SneakerActivity::class.java)

    @Test
    fun testFirstItemDetail() {
        KView(R.id.rvSneakerList) perform {
            clicOnItem(0)
        }

        KView(R.id.tvSneakerDetailName) check {
            hasText("Nike SF Air Force 1 Mid")
        }

        KView(R.id.tvSneakerDetailCollectiomName) check {
            hasText("Air Force 1")
        }
    }
}